frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToParent.java

```diff
@@ -119,15 +119,11 @@ private boolean handleMergedToParent(
 
     switch (slideOutDialog.show(project)) {
       case MessageConstants.YES :
+        // For a branch merged to its parent, we're not syncing to remote.
+        // Let's just go straight to the next branch.
+        Runnable doInUIThreadWhenReady = () -> graphTable.queueRepositoryUpdateAndModelRefresh(traverseNextEntry);
         new SlideOutBackgroundable(getString("action.GitMachete.BaseSlideOutAction.task.title"),
-            managedBranch, gitRepository, currentBranchIfManaged, branchLayout, graphTable) {
-          @Override
-          public void onSuccess() {
-            // For a branch merged to its parent, we're not syncing to remote.
-            // Let's just go straight to the next branch.
-            graphTable.queueRepositoryUpdateAndModelRefresh(traverseNextEntry);
```

Comment:
our gain here (probably not only one) is that `queueRepositoryUpdateAndModelRefresh` is not called again right after the call from `SlideOutBackgroundable`, am I right?